### PR TITLE
Fix: Bounds in output for Metadata::None variables

### PR DESCRIPTION
**Pull request by AstroBarker**
In updating `phoebus` to use up-to-date `parthenon` I ran into an issue where a `Metadata::None` variable was failing to output due to view bounds accesses. There seems to have been an off-by-one error in the relevant code block in `output_utils.hpp:313`. I've updated the bounds there to correctly output `Metadata::None` variables. 

To test this, I modified the advection example to have an optional `Metadata::None` variable (enabled in the input deck `<Advection>/test_metadata_none`). I set the output_hdf5 test to use this variable.

## PR Checklist

- [ ] Code passes cpplint
- [ ] New features are documented.
- [x] Adds a test for any bugs fixed. Adds tests for new features.
- [ ] Code is formatted
- [ ] Changes are summarized in CHANGELOG.md
- [ ] Change is breaking (API, behavior, ...)
  - [ ] Change is *additionally* added to CHANGELOG.md in the breaking section
  - [ ] PR is marked as breaking
  - [ ] Short summary API changes at the top of the PR (plus optionally with an automated update/fix script)
- [ ] CI has been triggered on [Darwin](https://re-git.lanl.gov/eap-oss/parthenon/-/pipelines) for performance regression tests.
- [ ] Docs build
- [ ] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

**Review by pdmullen — APPROVED** [review]
Nice find! Output bugs are always tricky to track down.

**Review by Yurlungur — APPROVED** [review]
Thanks for the fix! This breakage is definitely on me.

**Comment by Yurlungur** [comment]
Looks like tests are failing @AstroBarker ?

**Comment by AstroBarker** [comment]
> Looks like tests are failing @AstroBarker ?

Looks like `output_hdf5` so probably it's the gold files

**Comment by Yurlungur** [comment]
almost certainly. run locally and see what the error is. might need to update the gold files.

**Comment by AstroBarker** [comment]
> almost certainly. run locally and see what the error is. might need to update the gold files.

Yeah they pass locally for me only if I update the goldfiles. I added a new output for the `Metadata::None` variable in the hdf5 tests. What's the process for updating the proper gold files?

**Comment by Yurlungur** [comment]
Run `parthenon/tst/regression/gold_standard/make_tarball.sh`. Then create a release from the generated tarball. You will also need to update the tarball version and hash in the `CMake`

File: example/advection/advection_package.cpp
Comment (pgrete) [inline example/advection/advection_package.cpp]: I'm not sure what/why this is happening here.
If I read the the field correctly, then it's a vector with 3 components and extents larger than the mesh size by one.
In this `par_for` the block indices are used (without any global offset), so every block will have the same indices set to the same values and a large fraction of the index space will remain 0.
Is that intentional?

**Review by pgrete — COMMENTED** [review]
In addition to the comment on filling the content of the new var: Should we also add a proper check that this is working, i.e., read the data of the variable to confirm that the data written is looking what it's supposed to look like?

File: example/advection/advection_package.cpp
Comment (AstroBarker) [reply example/advection/advection_package.cpp]: Good catch, I didn't pay much mind to the contents of the fields, only that its shape reproduced the crash. This should be fixed to be filled now, I think.

File: example/advection/advection_package.cpp
Comment (pgrete) [inline example/advection/advection_package.cpp]: i just looked at the updated loop.
While I understand the fix going from mesh to meshbock, I now don't get the `ivar_lo`/`ivar_hi` indexing.
What's supposed to happen here? I could see this working if `ivar_lo`/`ivar_hi replace the `n` index.

File: tst/regression/test_suites/output_hdf5/parthinput.advection
Comment (pgrete) [inline tst/regression/test_suites/output_hdf5/parthinput.advection:69]: Given that this is now in the output file, how about reading the content to make sure that the data written is actually there.
I'm not sure we have that kind of test for `Meshdata::None` already.

File: example/advection/advection_package.cpp
Comment (pgrete) [reply example/advection/advection_package.cpp]: Any comment on this? ping @AstroBarker 

I'm happy to merge this without adding test to read the data (which would be great), but I'd like to understand this loop before merging as from my current understanding it might seem like being an unintended bug.

Comment (AstroBarker) [reply example/advection/advection_package.cpp]: I've updated the loop, removing the `ivar_lo/hi` stuff, which I agree was not necessary in the loop, and added a section to the hdf5 test that checks the contents of this variable. Sorry for the delay!

Comment (pgrete) [reply example/advection/advection_package.cpp]: Great, thanks!

**Review by pgrete — APPROVED** [review]

**Comment by pgrete** [comment]
Formatter does not work on forks. I'm force merging.

**Comment by pgrete** [comment]
I don't see the force merge button any more? Am I missing sth?

**Comment by Yurlungur** [comment]
> I don't see the force merge button any more? Am I missing sth?

I already clicked auto merge. It seems like not all the "long" tests triggered. I will try to disable/re-enable.

**Comment by pgrete** [comment]
> > I don't see the force merge button any more? Am I missing sth?
> 
> I already clicked auto merge. It seems like not all the "long" tests triggered. I will try to disable/re-enable.

Yes, that's what I did earlier today and all tests passed (except for the formatting one that, as we know, does not report results for PRs from forks).
Previously, there was a force merge button but I don't see it any more (maybe because of the changed branched protection rules?)

**Comment by Yurlungur** [comment]
> > > I don't see the force merge button any more? Am I missing sth?
> > 
> > 
> > I already clicked auto merge. It seems like not all the "long" tests triggered. I will try to disable/re-enable.
> 
> Yes, that's what I did earlier today and all tests passed (except for the formatting one that, as we know, does not report results for PRs from forks). Previously, there was a force merge button but I don't see it any more (maybe because of the changed branched protection rules?)

If you want to force merge, disable auto-merge. Then there should be an option.

**Comment by pgrete** [comment]
> > > > I don't see the force merge button any more? Am I missing sth?
> > > 
> > > 
> > > I already clicked auto merge. It seems like not all the "long" tests triggered. I will try to disable/re-enable.
> > 
> > 
> > Yes, that's what I did earlier today and all tests passed (except for the formatting one that, as we know, does not report results for PRs from forks). Previously, there was a force merge button but I don't see it any more (maybe because of the changed branched protection rules?)
> 
> If you want to force merge, disable auto-merge. Then there should be an option.

That is basically my question. I don't see the button any more.
So if you still see it, feel free to go ahead and force merge (as the tests including extended ones passed earlier today).

**Comment by Yurlungur** [comment]
> > > > > I don't see the force merge button any more? Am I missing sth?
> > > > 
> > > > 
> > > > I already clicked auto merge. It seems like not all the "long" tests triggered. I will try to disable/re-enable.
> > > 
> > > 
> > > Yes, that's what I did earlier today and all tests passed (except for the formatting one that, as we know, does not report results for PRs from forks). Previously, there was a force merge button but I don't see it any more (maybe because of the changed branched protection rules?)
> > 
> > 
> > If you want to force merge, disable auto-merge. Then there should be an option.
> 
> That is basically my question. I don't see the button any more. So if you still see it, feel free to go ahead and force merge (as the tests including extended ones passed earlier today).

Shoot you're right. I don't see it either. Maybe we submit a new MR on a branch and see if that goes through? Should figure out how to make fork merges work though...

**Comment by pgrete** [comment]
Alright, so it wasn't just me and it was related to the branch protection rules.
I recently enabled
image
because of my accidental push to `develop`.
So I temp disabled it, now force merged, and then re-enabled it as a quick workaround for this PR.

In the long run it'd definitely be nice to have the formatter also trigger for forks.
I looked into this in the past and was not able to figure out a solution within a reasonable time (and thus tabled it).
So from my point of view, I'm happy to continue working with a manual work around (unless more and more PRs are coming from forks) and spent the time debugging CI on other (CI) pieces ;)

**Comment by Yurlungur** [comment]
Aah I see. Ok sounds good.

**Pull request merged**
